File: Oxygene/Nougat/UIKit/UINavigationController/UINavigationController/SecondViewController.pas
namespace UINavigationController;

interface

uses
  UIKit;

type
  [IBObject]
  SecondViewController = public class(UIViewController)
  private
  public
    method init: id; override;
    method viewDidLoad; override;
    [IBAction] method buttonPressed(sender: id);
    method otherButtonPressed;
  end;

implementation

method SecondViewController.init: id;
begin
  self := inherited initWithNibName("SecondViewController") bundle(nil);
  if assigned(self) then begin
    self.title := "Second View Controller";
  end;
  result := self;
end;

method SecondViewController.buttonPressed(sender: id);
begin
  self.navigationController.popToRootViewControllerAnimated(true);
end;

method SecondViewController.otherButtonPressed;
begin
  var message := new UIAlertView withTitle("Search") message("If this were a real application you would now be in the DO STUFF mode") &delegate(nil) cancelButtonTitle("OK") otherButtonTitles(nil);
  message.show();
end;

method SecondViewController.viewDidLoad;
begin
  inherited viewDidLoad;
  var rightButton := new UIBarButtonItem withTitle("Do Stuff") style(UIBarButtonItemStyle.UIBarButtonItemStylePlain) &target(self) action(selector(otherButtonPressed)); 
  self.navigationItem.rightBarButtonItem := rightButton;
end;

end.

File: Hydrogene/Toffee/UIKit/UINavigationController/UINavigationController/SecondViewController.cs
using UIKit;

namespace UINavigationController
{
	[IBObject]
	public class SecondViewController: UIViewController
	{
		public override id init()
		{
			this = base.initWithNibName("SecondViewController") bundle(null);
      if (this != null) {
        this.title = "Second View Controller";
	    }
			return this;
		}

    public override void viewDidLoad()
    {
        base.viewDidLoad();

        UIBarButtonItem rightButton = new UIBarButtonItem withTitle("Do Stuff") style(UIBarButtonItemStyle.UIBarButtonItemStylePlain) target(this) action(__selector(otherButtonPressed));
        self.navigationItem.rightBarButtonItem  = rightButton;

    }

    [IBAction]
    public void buttonPressed(id sender)
    {
        this.navigationController.popToRootViewControllerAnimated(true);
    }

    public void otherButtonPressed()
    {
         UIAlertView message = new UIAlertView withTitle("Search Selected") message("If this were a real application you would now go into the DO STUFF mode") @delegate(null) cancelButtonTitle("OK") otherButtonTitles(null);
         message.show();
    }
	}
}